qa-pipelines/tasks/cf-deploy-upgrade-common.sh

```diff
@@ -60,23 +60,46 @@ is_namespace_ready() {
         | sort \
         | uniq) ]]
 }
+wait_for_jobs() {
+    local release=$1
+    local jobs_desired_remaining
+    start=$(date +%s)
+    for (( i = 0 ; i < 480 ; i ++ )); do
+        # Get the list of all jobs in the helm release, and subtract the value of 'completed' from 'desired'
+        # It would be better to parse this from `helm get manifest`, but since that command is broken in helm 
+        # v2.8.2 (https://github.com/helm/helm/issues/3833) for now we'll parse it from the human-readable status
+        jobs_desired_remaining=$(helm status $release | awk '/==> v1\/Job/ { getline; getline; while (NF>0) { print $2 - $3; getline } }')
```

Comment:
Alternatively, do the counting in awk:
```awk
BEGIN         { j=0; c=0 }
/==> v1\/Job/ { getline ; getline ; j=1 }
NF == 0       { j=0 }
j > 0         { c += $2 - $3 }
END           { print c }
```

Reply:
Ah, apparently it's possible to have completed > desired in some situations (kube just ensures that at least one succeeds). So we can't do what I have (unless we clamp `$2 - $3` to `0`), so taking the loop as-is with the `sort` is fine.